compiler/src/dotty/tools/dotc/core/TyperState.scala

```diff
@@ -253,7 +253,8 @@ class TyperState() {
       Stats.record("typerState.gc")
       val toCollect = new mutable.ListBuffer[TypeLambda]
       for tvar <- ownedVars do
-        assert(tvar.owningState.get eq this, s"Inconsistent state in $this: it owns $tvar whose owningState is ${tvar.owningState.get}")
+        val tvaros = tvar.owningState.nn.get
+        assert(tvaros != null && (tvaros eq this), s"Inconsistent state in $this: it owns $tvar whose owningState is ${tvaros}")
```

Comment:
The `!= null` condition is redundant, should be dropped.

Reply:
Done